pkg/compression/internal/types.go

```diff
@@ -35,6 +37,11 @@ func (c Algorithm) Name() string {
 	return c.name
 }
 
+// Name returns the MIME type to use for the compression algorithm.
+func (c Algorithm) MIME() string {
```

Comment:
The returned values are clearly not MIME types; this needs accurately thinking through what we are actually doing. What is the _precise_ semantics of the various fields, and the relationship between `zstd` and `zstd:chunked`?

Are they two completely distinct file formats (so that `DetectCompression` must differentiate between them?), two related file formats where on is-a the other (so that we can conservatively almost always use only one of them), or is “chunked” maybe just a compressor algorithm option like `types.SystemContext.CompressionLevel` that _only_ matters for creating streams? Something else?

Reply:
they are two related file formats.  The zstd decompressor can handle `zstd:chunked` but it loses some of its features.

In future we'd probably need the same differentiation between gzip and stargz.  Where gzip can handle the stargz format but it won't handle its additional features

Reply:
Yes; what I’d like to see is a precise modeling and correct naming of the individual functions/values — so that any future updates to the code can tell what was the purpose, and what are the invariants that need to be maintained, without understanding the complete codebase.

(Or a very clear one-time exception — the original version had a `strings.Split(algorithm.Name(), ":")` somewhere in the compression/MIME mapping code. The concern with that was essentially the same as with the above: introducing a data model with undocumented and unclear semantics — and at that place a specific `if algorithm.Name() == "zstd:chunked"` was one of the options discussed.

While exceptions are worse than an ideal clean code that can predict future needs, it’s fine if we don’t know what is a good long-term model, and make a very explicit, clearly delineated, and reasonably documented, one-time exception.)

Reply:
@giuseppe Any response to this comment?